inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_base.cpp

```diff
@@ -207,12 +207,24 @@ KernelsData ConvolutionKernelBase::GetCommonKernelsData(const Params& params,
                                                         const optional_params& options,
                                                         const std::string exeMode,
                                                         int autoTuneIndex) const {
-    if (!Validate(params, options)) {
+    KernelData kd = KernelData::Default<convolution_params>(params);
+    convolution_params& newParams = *static_cast<convolution_params*>(kd.params.get());
+
+    bool succeed = UpdateWeightsParams(newParams,
```

Comment:
Won't this change affect some other kernels? As I understand, after this patch each kernel will have transformed (and aligned) weights tensor in SetDefault method, so maybe the behavior of the kernels that use weights layout will change. 

Maybe it'd more safe to update only imad kernel and use output tensor instead of weights in this method?

Reply:
Before this change weights.OFM().v == output.Feature().v, which caused not intended growth of global work sizes.
Computation time of kernels using weights.OFM().v, should be now improved by setting intended OFM values . 
If any kernel will be affected, solution would be to change weights.OFM().v to output.Feature().v. 
weights.IFM().v shouldn't be affected by my change in any case.